### PR TITLE
DeepCopy rework

### LINKED ISSUE #436: [FEATURE] Remove TlaDecl.deepCopy() and TlaEx.deepCopy()
I just looked for usages of `TlaDecl.deepCopy` and `TlaEx.deepCopy()`. There are less than 10 lines of code that use them.

I find `deepCopy` problematic, as it does not cooperate with `TransformationTracker`. Shall we remove them? For now, I have marked them as deprecated.

### PR DESCRIPTION
**Pull request by Kukovec**
closes  #436

After #444 passes, `deepCopyDeclInternal` should be updated to a declaration transformation.

### CODE REVIEW

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/DeepCopy.scala
Comment (konnov) [inline tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/DeepCopy.scala:9]: why do you introduce `deepCopyEx` instead of implementing `TlaExTransformation`? You could implement both `TlaExTransformation` and `TlaDeclTransformation`. (I am going to merge #444 as soon as it passes all the tests).

**Review by konnov — COMMENTED** [review]
Looks great! But I am confused about deepCopyEx, see the comment.

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/DeepCopy.scala
Comment (Kukovec) [reply tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/DeepCopy.scala:9]: I was debating this internally. The reason I went in this direction is exactly that: `DeepCopy` would extend two transformations and therefore have two overloaded `apply` methods, which you couldn't visually tell apart in use cases. This implementation, I think, makes it impossible to confuse the two methods.

**Comment by konnov** [comment]
I am about to merge #444. You will be able to merge this PR after that.

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/DeepCopy.scala
Comment (konnov) [reply tlair/src/main/scala/at/forsyte/apalache/tla/lir/transformations/standard/DeepCopy.scala:9]: I see. The `apply` methods may be confusing.

**Review by konnov — APPROVED** [review]
LGTM

**Comment by konnov** [comment]
Let's include it in the next release. Please do not merge now.

**Comment by Kukovec** [comment]
> 
> 
> Let's include it in the next release. Please do not merge now.

Sure, just say when.

**Comment by konnov** [comment]
It should be ok. The release is out.

**Pull request merged**
